lib/Lib.IntTypes.fsti

```diff
@@ -1,705 +1,817 @@
+  -> a:int_t t l 
+  -> b:int_t t l 
+  -> m:nat{m < bits t} 
+  -> Lemma
+    (requires 0 <= v a /\ v a < pow2 m /\ v b % pow2 m == 0)
+    (ensures  v (a `logor` b) == v a + v b)
+  //[SMTPat (v (a `logor` b))]
 
 inline_for_extraction
-val lognot: #t:inttype -> #l:secrecy_level
-  -> a:uint_t t l
-  -> uint_t t l
+val lognot: #t:inttype -> #l:secrecy_level -> int_t t l -> int_t t l
 
 inline_for_extraction
-type shiftval (t:inttype) = u:size_t{uint_v u < bits t}
+type shiftval (t:inttype) = u:size_t{v u < bits t}
 
 inline_for_extraction
-type rotval  (t:inttype) = u:size_t{uint_v u > 0 /\ uint_v u < bits t}
+type rotval (t:inttype) = u:size_t{0 < v u /\ v u < bits t}
 
 inline_for_extraction
 val shift_right: #t:inttype -> #l:secrecy_level
-  -> a:uint_t t l
-  -> b:shiftval t
-  -> c:uint_t t l
+  -> int_t t l
+  -> shiftval t
+  -> int_t t l
 
 val shift_right_lemma: #t:inttype -> #l:secrecy_level
-  -> a:uint_t t l
+  -> a:int_t t l
   -> b:shiftval t
   -> Lemma
-    (uint_v #t #l (shift_right #t #l a b) ==  uint_v #t #l a / pow2 (uint_v #U32 #PUB b))
-    [SMTPat (uint_v #t #l (shift_right #t #l a b))]
+    (v (shift_right a b) == v a / pow2 (v b))
+    [SMTPat (v #t #l (shift_right #t #l a b))]
 
 inline_for_extraction
 val shift_left: #t:inttype -> #l:secrecy_level
-  -> a:uint_t t l
-  -> b:shiftval t
-  -> c:uint_t t l
-
-val shift_left_lemma: #t:inttype -> #l:secrecy_level
-  -> a:uint_t t l
+  -> int_t t l
+  -> shiftval t
+  -> int_t t l
+
+val shift_left_lemma: 
+    #t:inttype 
+  -> #l:secrecy_level
+  -> a:int_t t l
   -> b:shiftval t
   -> Lemma
-    (uint_v #t #l (shift_left #t #l a b) == (uint_v #t #l a `op_Multiply` pow2 (uint_v #U32 #PUB b)) % modulus t)
-    [SMTPat (uint_v #t #l (shift_left #t #l a b))]
+    (v (shift_left a b) == (v a * pow2 (v b)) @%. t)
+    [SMTPat (v #t #l (shift_left #t #l a b))]
 
 inline_for_extraction
 val rotate_right: #t:inttype -> #l:secrecy_level
-  -> a:uint_t t l
-  -> b:rotval t
-  -> uint_t t l
+  -> int_t t l
+  -> rotval t
+  -> int_t t l
 
 inline_for_extraction
 val rotate_left: #t:inttype -> #l:secrecy_level
-  -> a:uint_t t l
-  -> b:rotval t
-  -> uint_t t l
+  -> int_t t l
+  -> rotval t
+  -> int_t t l
 
 ///
 /// Masking operators for all machine integers
 ///
 
 inline_for_extraction
-val zeroes: t:inttype -> l:secrecy_level -> v:uint_t t l{uint_v v = 0}
-
-inline_for_extraction
-val ones: t:inttype -> l:secrecy_level -> v:uint_t t l{uint_v v = maxint t}
-
-inline_for_extraction
-val eq_mask: #t:inttype -> uint_t t SEC -> uint_t t SEC -> uint_t t SEC
+val eq_mask: #t:inttype{unsigned t} -> int_t t SEC -> int_t t SEC -> int_t t SEC
 
-val eq_mask_lemma: #t:inttype -> a:uint_t t SEC -> b:uint_t t SEC -> Lemma
-  (requires True)
-  (ensures  (v a = v b ==> v (eq_mask a b) == maxint t) /\
-            (v a <> v b ==> v (eq_mask a b) == 0))
+val eq_mask_lemma: #t:inttype{unsigned t} -> a:int_t t SEC -> b:int_t t SEC -> Lemma
```

Comment:
Same, this requires `unsigned t` so the proof given is actually not proving anything about signed types.